tests/rados/test_reads_balancer.py

```diff
@@ -0,0 +1,294 @@
+
+                    time.sleep(5)
+                    # Checking if it is possible to revert the changes made previously via "pg-upmap-primary"
+                    rm_cmd = f"ceph osd rm-pg-upmap-primary {pgid}"
+                    out, err = client_node.exec_command(sudo=True, cmd=rm_cmd)
+                    # Checking the change in the new acting set
+                    time.sleep(20)
+                    final_acting_set = rados_obj.get_pg_acting_set(pg_num=pgid)
+                    if final_acting_set[0] != acting_set[0]:
+                        log.error(
+                            f"The acting set has not reverted to original."
+                            f"Old AC : {acting_set}, final AC : {final_acting_set}"
+                        )
+                        raise Exception("AC not changed error")
+
+                    log.info("Completed tests on replicated pools for online commands")
+                    break
+                else:
+                    log.info(f"Selected pool {pool_name} is a EC pool, skipping")
+
+            for pool_name in existing_pools:
+                pool_details = rados_obj.get_pool_details(pool=pool_name)
+                log.debug(f"Selected pool name : {pool_name}")
+                # Testing on EC pools
+                if pool_details["erasure_code_profile"]:
+                    pool = pool_details["pool_name"]
+
+                    # Fetching the acting set for the pool
+                    # Getting a sample PG from the pool
+                    pool_id = pool_obj.get_pool_id(pool_name=pool)
+                    pgid = f"{pool_id}.0"
+                    acting_set = rados_obj.get_pg_acting_set(pg_num=pgid)
+
+                    log.debug(
+                        f"Acting set for PG-id {pgid} in pool {pool} is {acting_set}"
+                    )
+
+                    # Trying to change Primary PG for a EC pool. should not be possible. -ve test
+                    new_osd = acting_set[1]
+                    change_cmd = f"ceph osd pg-upmap-primary {pgid} {new_osd}"
+                    error_str = "Error EINVAL: pg-upmap-primary is only supported for replicated pools"
+                    try:
+                        out, err = client_node.exec_command(sudo=True, cmd=change_cmd)
+                    except Exception as e:
+                        if error_str not in str(e).strip():
+                            log.error(
+                                "Correct error string not seen in stderr stream"
+                                f"Expected : {error_str} \n Got : {e}"
+                            )
+                            raise Exception("Wrong error string obtained")
+                        log.error(f"An error occurred but was expected: {e}")
+                        log.debug("Could not change the primary for EC pools. Pass")
+                    break
+                log.info("Verified the Online commands to change acting set")
+                return 0
```

Comment:
Very minor, these two lines should be outside the for loop starting at line #248